addons/binding/org.openhab.binding.vitotronic/src/main/java/org/openhab/binding/vitotronic/internal/discovery/VitotronicBridgeDiscovery.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.vitotronic.internal.discovery;
+/**
+ * The {@link VitotronicBridgeDiscovery} class handles the discovery of optolink adapter
+ * with broadcasting and put it to inbox, if found.
+ * 
+ * 
+ * @author Stefan Andres - Initial contribution
+ */
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.vitotronic.VitotronicBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class VitotronicBridgeDiscovery extends AbstractDiscoveryService  {
+	
+	private final static Logger logger = LoggerFactory.getLogger(VitotronicBridgeDiscovery.class);
+
+	public VitotronicBridgeDiscovery() throws IllegalArgumentException {
+		super(VitotronicBindingConstants.SUPPORTED_BRIDGE_THING_TYPES_UIDS, 15);
+	}
+
+	@Override
+	protected void startScan() {
+		logger.trace("Start discovery of Vitotronic Optolink Adapter (VOP)");
+		searchVitotronicAdapter(VitotronicBindingConstants.BROADCAST_PORT);
+		
+	}
+	
+	
+	@Override
+	protected void startBackgroundDiscovery() {
+		// do nothing
+	}
+
+	@Override
+	public boolean isBackgroundDiscoveryEnabled() {
+		return false;
+	}
+	
+	
+	private void searchVitotronicAdapter(final int port) {
+		// if found returns IP Address else null
+		logger.trace("Start broadcast discovery Thread)");
+		
+		// Go into background for not blocking openhab core 
+
+		Thread thread = new Thread("Discovery") {

```

Comment:
Thanks. Good feature. I going to change it to:
`scheduler.execute(searchRunnable);`